include/fiction/algorithms/simulation/sidb/occupation_probability_of_excited_states.hpp

```diff
@@ -29,21 +29,25 @@ namespace fiction
  * @param temperature System temperature to assume.
  * @return The occupation probability of all erroneous states is returned.
  */
-[[nodiscard]] inline double occupation_probability_gate_based(const energy_and_state_type& energy_distribution,
-                                                              const double                 temperature) noexcept
+[[nodiscard]] inline double occupation_probability_gate_based(const sidb_energy_and_state_type& energy_distribution,
+                                                              const double                      temperature) noexcept
 {
-    assert(!energy_distribution.empty() && "vector is empty");
     assert((temperature > static_cast<double>(0)) && "temperature should be slightly above 0 K");
 
+    if (energy_distribution.empty())
+    {
+        return 0.0;
+    }
+
     auto min_energy = std::numeric_limits<double>::infinity();
 
     // Determine the minimal energy.
-    for (const auto& [energy, degeneracy] : energy_distribution)
+    if (!energy_distribution.empty())
```

Comment:
This case you have already ruled out by the condition above and don't need to query again.